Relese

YousraMamdouh opened the pull request.

YousraMamdouh committed:
this commit will :
- creating CustomCell
- adding homeViewController with(tableView,CollectionView)

mina249 committed:
add network and home view model logic

YousraMamdouh committed:
Merge branch 'yousra' into relese

# Conflicts:
#	FoodRecipes.xcodeproj/project.pbxproj
#	FoodRecipes/ViewController.swift

Ayahamdy97 commented on FoodRecipes/Constants.swift:
```diff
@@ -0,0 +1,11 @@
+//
+
+import Foundation
+struct K{
```
Rename struct with meaningful name

mina249 replied:
done

Ayahamdy97 commented on FoodRecipes/Models/MealModel.swift:
```diff
@@ -0,0 +1,18 @@
+//
+
+import Foundation
+class MealModel:Codable{
```
Separate classes in different files

mina249 replied:
done

Ayahamdy97 commented on FoodRecipes/Models/MealModel.swift:
```diff
@@ -0,0 +1,18 @@
+}
+class Meal:Codable{
+    let name:String?
+    let show_id:Int?
```
don't use "_" while naming functions

mina249 replied:
done

Ayahamdy97 commented on FoodRecipes/Models/MealModel.swift:
```diff
@@ -0,0 +1,18 @@
+    let name:String?
+    let show_id:Int?
+    let num_servings:Int?
+    
```
remove extra spaces

mina249 replied:
done

Ayahamdy97 commented on FoodRecipes/Modules/Home/View/CategoriesCell/CategoriesCellTableViewCell.swift:
```diff
@@ -0,0 +1,36 @@
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
+         cellView.layer.cornerRadius = 20
```
Place code in function

Ayahamdy97 commented on FoodRecipes/Modules/Home/View/HomeViewController.swift:
```diff
@@ -0,0 +1,72 @@
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
```
remove commented code

Ayahamdy97 commented on FoodRecipes/Modules/Home/View/HomeViewController.swift:
```diff
@@ -0,0 +1,72 @@
+        super.viewDidLoad()
+        collectionView.register(UINib(nibName: "TopCell", bundle: nil), forCellWithReuseIdentifier: "cell")
+        tableView.register(UINib(nibName: "CategoriesCellTableViewCell", bundle: nil), forCellReuseIdentifier: "CategoriesCellTableViewCell")
+        // Do any additional setup after loading the view.
```
dont place code in viewDidLoad, use functions!

Ayahamdy97 commented on FoodRecipes/Modules/Home/View/HomeViewController.swift:
```diff
@@ -0,0 +1,72 @@
+import UIKit
+
+class HomeViewController: UIViewController {
+    let categories: [Category] = [Category(name: "Popular", image: "fire"),Category(name: "Breakfast", image: "breakfast"),Category(name: "Lunch", image: "lunch-box"),Category(name: "Dinner", image: "dinner"),Category(name: "Dessert", image: "cake")]
```
place array in view model

Ayahamdy97 commented on FoodRecipes/Modules/Home/View/HomeViewController.swift:
```diff
@@ -0,0 +1,72 @@
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as! TopCell
+        cell.categoryName.text = categories[indexPath.row].name
```
Make function inside cell that setups UI

Ayahamdy97 commented on FoodRecipes/Modules/Home/View/TopCustomCell/TopCell.swift:
```diff
@@ -0,0 +1,22 @@
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
+        backView.alpha = 1.0
```
Use Function

Ayahamdy97 commented on FoodRecipes/Modules/Home/View/TopCustomCell/TopCell.swift:
```diff
@@ -0,0 +1,22 @@
+    }
+
+    @IBOutlet weak var backView: UIView!
+    @IBOutlet weak var categoryImage: UIImageView!
```
Place outlets at the beginning of the class

Ayahamdy97 commented on FoodRecipes/Services/NetworkManager/Network.swift:
```diff
@@ -0,0 +1,40 @@
+import Alamofire
+
+class Network{
+    
```
remove extra spaces

mina249 replied:
done

Ayahamdy97 commented on FoodRecipes/Utilities/Style.swift:
remove unwanetd comments

Mariam-Adly committed:
add custom cell

YousraMamdouh committed:
in this commit:
- taking the response from viewModel and setting data in view

YousraMamdouh committed:
Merge branch 'relese' into mariam

# Conflicts:
#	FoodRecipes.xcodeproj/project.pbxproj
#	FoodRecipes/ViewController.swift

mina249 committed:
editing tabbar

YousraMamdouh committed:
in this commit:
- changing category cell background if selected
- retrieving data from API according to category selected

YousraMamdouh committed:
Merge branch 'relese' into Rendring-Data

# Conflicts:
#	FoodRecipes.xcodeproj/project.pbxproj
#	FoodRecipes/CategoriesCell/CategoriesCellTableViewCell.swift

mina249 committed:
fix conflict

mina249 committed:
Merge remote-tracking branch 'refs/remotes/origin/relese'

mina249 committed:
fix conflict

mina249 committed:
Merge branch 'relese'

Conflicts:
	FoodRecipes.xcodeproj/project.pbxproj
	FoodRecipes/Base.lproj/Main.storyboard
	FoodRecipes/SceneDelegate.swift
	Pods/Pods.xcodeproj/project.pbxproj

mina249 committed:
this commit will :
edit .gitignore

mina249 committed:
connecting screens together

YousraMamdouh committed:
in this commit:
- UI of home screen enhanced
- animation view added to home screen
- fetching details of specific meal using Alamofire
- retrieving data of simillar meals from API

YousraMamdouh committed:
this commit :
change .getignore file

YousraMamdouh committed:
this commit will :
change .gitignore

YousraMamdouh committed:
Merge branch 'relese' into details_screen

# Conflicts:
#	.gitignore

mina249 committed:
adding details ScreenUI

YousraMamdouh committed:
Merge branch 'relese' into DetailsScreenUI

# Conflicts:
#	FoodRecipes.xcodeproj/project.pbxproj

YousraMamdouh committed:
in this commit:
merging

YousraMamdouh committed:
in this commit:
enhancing tab bar
